source/agent/addons/avstreamLib/AVStreamOutWrap.cc

```diff
@@ -120,10 +120,10 @@ void AVStreamOutWrap::New(const v8::FunctionCallbackInfo<v8::Value>& args)
             opts.hls_list_size = Nan::To<int32_t>(
                 Nan::Get(parameters, Nan::New("hlsListSize").ToLocalChecked()).ToLocalChecked()).FromJust();
 
-            strncpy(opts.hls_method,
-                    getString(Nan::Get(parameters, Nan::New("method").ToLocalChecked()).ToLocalChecked()).c_str(),
-                    sizeof(opts.hls_method) - 1);
-            opts.hls_method[sizeof(opts.hls_method) - 1] = '\0';
+            const char *method_str = getString(Nan::Get(parameters, Nan::New("method").ToLocalChecked()).ToLocalChecked()).c_str() ;
+            uint32_t s = strlen(method_str) < sizeof(opts.hls_method) ? strlen(method_str) : sizeof(opts.hls_method) - 1 ;
+            memset(opts.hls_method, 0, sizeof(opts.hls_method));
+            strncat(opts.hls_method, method_str ,s );
```

Comment:
Could you use `sizeof(opts.hls_method) - 1` instead of `s` here?
And remove space ` ` before `)` or `;`